influx/query.go

```diff
@@ -0,0 +1,56 @@
+package influx
+
+import (
+	"context"
+	"crypto/x509"
+	"fmt"
+
+	"github.com/apache/arrow/go/v12/arrow/flight/flightsql"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/metadata"
+)
+
+func (c *Client) initializeQueryClient() error {
+	url, safe := ReplaceURLProtocolWithPort(c.configs.HostURL)
+
+	var transport grpc.DialOption
+
+	if (safe == nil || *safe) {
+		pool, err := x509.SystemCertPool()
+		if err != nil {
+			return fmt.Errorf("x509: %s", err)
+		}
+		transport = grpc.WithTransportCredentials(credentials.NewClientTLSFromCert(pool, ""))
+	} else {
+		transport = grpc.WithTransportCredentials(insecure.NewCredentials())
+	}
+
+	opts := []grpc.DialOption{
+		transport,
+	}
+
+	client, err := flightsql.NewClient(url, nil, nil, opts...)
+	if err != nil {
+		return fmt.Errorf("flightsql: %s", err)
+	}
+	c.queryClient = client
+	return nil
+}
+
+func (c *Client) Query(ctx context.Context, database string, query string, queryParams interface{}) (*QueryIterator, error) {
+	ctx = metadata.AppendToOutgoingContext(ctx, "authorization", "Bearer "+c.configs.AuthToken)
+	ctx = metadata.AppendToOutgoingContext(ctx, "bucket-name", database)

```

Comment:
The parameter should be "database":

```suggestion
	ctx = metadata.AppendToOutgoingContext(ctx, "database", database)

```

Reply:
currently `bucket-name` required and `database` doesn't work

Reply:
I am tested it with new account at "https://us-east-1-1.aws.cloud2.influxdata.com" and "database" works correctly.

Reply:
On my side works only bucket-name. I`m proposing to keep both bucket-name and database for compatability reasons until interfaces unify.

Reply:
I am ok with this. I will also update C# client...